### PR TITLE
[TRTLLM-10666][chore] Refactor request fetching logic for better separation of concerns

### PR DESCRIPTION
**Pull request by lancelly**
### Summary

This PR refactors the request fetching and queue management logic, separating responsibilities between `ExecutorRequestQueue` and `PyExecutor`. This is the **first step** of a larger refactoring effort — in follow-up PRs, `waiting_queue` management will be moved to the new scheduler.

#### 1. `ExecutorRequestQueue` — Simplified to pure queue operations

- Focused on `request_queue` management only

#### 2. `PyExecutor` — Takes over request processing orchestration

- Owns `waiting_queue` (previously in `ExecutorRequestQueue`)
- Restructured `_fetch_new_requests` as unified entry point:

| Step | Method | Description |
|------|--------|-------------|
| 1 | `_fetch_and_enqueue_requests` | fetch + broadcast + enqueue |
| 2 | `_select_requests_for_tp` / `_select_requests_for_dp` | select from waiting queue |
| 3 | `_update_new_active_requests_queue_latency` | update performance metrics |
| 4 | `merge_requests` | convert to LlmRequest |

#### 3. `request_utils.py` — Extracted pure functions

- `get_from_waiting_queue`
- `merge_requests`
- `attach_py_objects_to_requests`

### Next Steps

- Move `waiting_queue` management to the new scheduler
- Refactor IFB schedule flow to:
  1. `schedule_local_active` — Schedule existing active requests
  2. `fetch_new_requests_to_waiting_queue` — Fetch and enqueue new requests
  3. `assign_waiting_requests` — Assign requests from waiting queue
  4. `schedule_new_active_requests` — Schedule newly activated requests

## Release Notes

* **Refactor**
  * Simplified request queue architecture and operations for cleaner request handling
  * Restructured request scheduling and distribution workflows across distributed ranks

* **New Features**
  * Enhanced request scheduling and balancing across tensor-parallel processing units
  * Improved latency tracking and performance metrics for queued requests
  * New request broadcasting and synchronization capabilities for multi-rank environments

* **Tests**
  * Added comprehensive test coverage for request utilities and executor operations

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: tensorrt_llm/_torch/pyexecutor/executor_request_queue.py [hunk 1]
```diff
@@ -1,24 +1,17 @@
...
 import datetime
-import heapq
-import os
 import queue
 import threading
 import time
-from collections import deque, namedtuple
 from itertools import repeat
-from typing import Dict, Iterable, List, Optional, Tuple
+from typing import Iterable, List, Optional, Tuple
```
[the remark this answers is not in the document]
Comment (lancelly) [reply tensorrt_llm/_torch/pyexecutor/executor_request_queue.py]: The files under directory pyexecutor all do not contain this header.

File: tensorrt_llm/_torch/pyexecutor/py_executor.py [hunk 2]
```diff
@@ -2016,6 +2040,180 @@ def _validate_request(self, request: LlmRequest):
...
+    def _update_new_active_requests_queue_latency(
+            self, new_requests: List[RequestQueueItem]):
+        """Update queue latency metrics for new requests."""
+        now = time.time()
+        latency = self.executor_request_queue.calculate_queue_latency(
+            new_requests, now)
+        self.new_active_requests_queue_latency_ms += latency
+
+    def _get_new_active_requests_queue_latency(self) -> float:
+        return self.new_active_requests_queue_latency_ms
```
[the remark this answers is not in the document]
Comment (lancelly) [reply tensorrt_llm/_torch/pyexecutor/py_executor.py]: This is alinged with the previous implementation.

File: tensorrt_llm/_torch/pyexecutor/py_executor.py [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import threading
 import time
 import traceback
+from collections import deque
 from contextlib import contextmanager
 from typing import Callable, Dict, Iterable, List, Optional, Tuple, Union
```
[the remark this answers is not in the document]
Comment (lancelly) [reply tensorrt_llm/_torch/pyexecutor/py_executor.py]: No need.

File: tensorrt_llm/_torch/pyexecutor/py_executor.py [hunk 4]
```diff
@@ -2026,6 +2049,182 @@ def _validate_request(self, request: LlmRequest):
...
+
+        # 6. Merge requests
+        return merge_requests(new_requests,
+                              cp_config=self.dist.cp_config,
+                              cp_rank=self.dist.cp_rank,
+                              cp_size=self.dist.cp_size,
+                              exclude_last_generation_logits=self.
+                              _should_exclude_last_generation_logits())
+
+    def _schedule_attention_dp_requests(
```
Comment (QiJune) [inline tensorrt_llm/_torch/pyexecutor/py_executor.py]: This function seems too simple, which just calls `schedule_attention_dp_requests`. So, is it reasonable to remove it?

**Review by QiJune — APPROVED** [review]
LGTM

File: tensorrt_llm/_torch/pyexecutor/py_executor.py (hunk 4 above)
Comment (lancelly) [reply tensorrt_llm/_torch/pyexecutor/py_executor.py]: Thanks for the review!
Sure, you're right. We can either remove this method in a follow-up PR, or refactor it to abstract out the request distribution interface for DP . If there are no other issues with this PR, could we merge it first? Running CI is quite time-consuming...

**Review by litaotju — APPROVED** [review]
